### ISSUE TITLE
Anonymous default-exported functions are skipped by the parser

### ISSUE THREAD

**Issue by fabian-barney** [fabian-barney/crap-typescript#4]
`parseFileMethods` drops `FunctionDeclaration` nodes when `node.name` is missing. That excludes anonymous default-exported functions even though they are real function bodies that should participate in CRAP analysis.

Reproduction:
1. Parse a file containing:

   ```ts
   export default function () {
     if (true) {
       return 1;
     }
     return 0;
   }
   ```
2. `parseFileMethods(file)` returns `[]`.

Actual behavior:
- The file contributes no metrics at all.

Expected behavior:
- Anonymous default-exported function declarations should be reported with a stable synthetic name (for example `default`) instead of being ignored.

Why this matters:
- Default-exported functions are common in application code. Skipping them under-reports both complexity and CRAP scores for the analyzed module.

Relevant code:
- `packages/core/src/parser.ts:26-30`
